ledger/ledger.go

```diff
@@ -277,40 +305,41 @@ func (l *Ledger) verifyMatchingGenesisHash() (err error) {
 	return
 }
 
-func openLedgerDB(dbPathPrefix string, dbMem bool, cfg config.Local, log logging.Logger) (trackerDBs trackerdb.Store, blockDBs db.Pair, err error) {
-	// Backwards compatibility: we used to store both blocks and tracker
-	// state in a single SQLite db file.
-	if !dbMem {
-		commonDBFilename := dbPathPrefix + ".sqlite"
-		_, err = os.Stat(commonDBFilename)
-		if !os.IsNotExist(err) {
-			// before launch, we used to have both blocks and tracker
-			// state in a single SQLite db file. We don't have that anymore,
-			// and we want to fail when that's the case.
-			err = fmt.Errorf("a single ledger database file '%s' was detected. This is no longer supported by current binary", commonDBFilename)
-			return
-		}
+func openLedgerDB(dbPrefixes DirsAndPrefix, dbMem bool, cfg config.Local, log logging.Logger) (trackerDBs trackerdb.Store, blockDBs db.Pair, err error) {
+	outErr := make(chan error, 2)
+
+	// resolve paths to tracker and block databases
+	trackerDBPrefix := filepath.Join(dbPrefixes.ResolvedGenesisDirs.RootGenesisDir, dbPrefixes.DBFilePrefix)
+	blockDBPrefix := filepath.Join(dbPrefixes.ResolvedGenesisDirs.RootGenesisDir, dbPrefixes.DBFilePrefix)
+	if dbPrefixes.ResolvedGenesisDirs.HotGenesisDir != "" {
+		trackerDBPrefix = filepath.Join(dbPrefixes.ResolvedGenesisDirs.HotGenesisDir, dbPrefixes.DBFilePrefix)
+	}
+	if dbPrefixes.ResolvedGenesisDirs.TrackerGenesisDir != "" {
+		trackerDBPrefix = filepath.Join(dbPrefixes.ResolvedGenesisDirs.TrackerGenesisDir, dbPrefixes.DBFilePrefix)
+	}
+	if dbPrefixes.ResolvedGenesisDirs.ColdGenesisDir != "" {
+		blockDBPrefix = filepath.Join(dbPrefixes.ResolvedGenesisDirs.ColdGenesisDir, dbPrefixes.DBFilePrefix)
+	}
+	if dbPrefixes.ResolvedGenesisDirs.BlockGenesisDir != "" {
+		blockDBPrefix = filepath.Join(dbPrefixes.ResolvedGenesisDirs.BlockGenesisDir, dbPrefixes.DBFilePrefix)
 	}
 
-	outErr := make(chan error, 2)
 	go func() {
 		var lerr error
 		switch cfg.StorageEngine {
 		case "sqlite":
 			fallthrough
 		// anything else will initialize a sqlite engine.
 		default:
-			file := dbPathPrefix + ".tracker.sqlite"
-			trackerDBs, lerr = sqlitedriver.Open(file, dbMem, log)
+			trackerDBs, lerr = sqlitedriver.Open(trackerDBPrefix+".tracker.sqlite", dbMem, log)
```

Comment:
I'll want to make sure that as the pebble implementation is checked in, it also uses `trackerDBPrefix`, since I renamed it away from "file"